aip/general/0213.md

```diff
@@ -9,44 +9,89 @@ placement:
+have particular requirements of how financial values are represented, leading
+to multiple finance-oriented organization-specific common components - because
+any global common component would either not meet the organization-specific
+requirements, or be too complex for general use.
 
-## Existing common protos
+## Guidance
 
-The common protos, which public-facing protos for an API **may** safely import,
-are as follows:
+- Organization-wide common component packages **must** end with `.type`, e.g.
+  `google.geo.type` or `google.shopping.type`.
+- Organizations **must** consult the API design team before creating a new
+  organization-wide common component package.
+- Organizations **must not** define generic components in organization-specific
+  common component packages, instead preferring global common components.
+- Common components **must not** be "moved" from an organization-specific common
+  component package to a global common package or vice versa.
+- Fields **should not** be added to existing messages.
+- Values **should not** be added to existing enums.
+- Fields **must not** be removed from existing messages.
+- Values **must not** be removed from existing enums.
+- While documentation **may** be clarified, it **should not** change the
+  meanings of existing values, including the validity of any given message
+  or set of messages.
+- New proto messages and enums **may** be added to common component packages.
+  - API teams  **should** allow sufficient time for propagation to clients
+    before using the new messages and enums in their APIs. Fields may take some
+    time for any changes to propagate through publication to client libraries
+    and other surfaces.
+  - API teams **should** consult widely within their organization, and ideally
+    with the API design team, before adding a new message or enum, due to the
+    limitations listed above.
+
+## Rationale
+
+Common components are effectively unversioned: APIs evolve independently of
+each other, both in terms of definition and implementation. A change such as
+adding a field is backward-compatible and predictable in specific APIs, and the
+API team can ensure that the server implementation is available before the API
+definition is published. By contrast, a change in a common component would
+effectively be universally available even if most API implementations did not
+take it into account.
+
+Adding a new message or enum is backward-compatible, as it does not affect
+existing APIs that may import other messages or enums from the same common
+component package.
+
+Consultation with the API design team is required for global common components
+and suggested for organization-specific common components as the border between
+"generic" and "organization-specific" is a gray area; some generic *concepts*
+have organization-specific use cases which surface through the components.
+
+## Existing global common components
```

Comment:
I'd love to see a section for `Existing Organization-wide common components` as well. I understand that this runs the risk of going out of date, but it would still be very helpful to categorize the ones that exist today for reference.

Reply:
Okay - I'm basically on vacation now until next week, and I'd like to get this merged before then if possible, but I can definitely take this as an action item for next week.

Reply:
SGTM! Added an issue to follow up https://github.com/aip-dev/google.aip.dev/issues/1142